### ISSUE TITLE
[CTA]: Expose additional icon types to `cta-type` attribute

### ISSUE THREAD

**Issue by proeung** [carbon-design-system/carbon-for-ibm-dotcom#8785]
**Engineering info:**  
- Innovation Team engineer: 
- Innovation Team FE JIRA Ticket: https://jsw.ibm.com/browse/HC-2482
- AEM BE JIRA Ticket: https://jsw.ibm.com/browse/ADCMS-2283
- DPO consulting engineer: N/A

----------------

### The problem

- Currently, the `cta-type` attribute/prop supports the following SVG icons
  - Local
  - Jump
  - External
  - Download
  - Video
  
<img width="1172" alt="Screen Shot 2022-05-03 at 1 51 54 PM" src="https://user-images.githubusercontent.com/1815714/166525682-24d1ecb4-2305-437e-805d-b5f476fe0245.png">

- However, within the AEM CMS, we need to be able to ultize SVG icons for `PDF` and `Blog` content (see attached)

<img width="502" alt="image (78)" src="https://user-images.githubusercontent.com/1815714/166525765-d8e633d6-2b70-4fb1-8044-38a8a83c6c16.png">

### The solution

- There are certain options within the AEM CMS dropdown that are inconsistent with the options/labels that are in the dotcom library and I’d like to use this issue as a way to help add missing icons and help AEM adjust their naming convention to align to the current `cta-type`  attribute (eg. `PDF` should be `Download`).
- Expose Carbon's Blog SVG icon or any other icons in the CTA type rendering.
- @oliviaflory @RichKummer I'd like to see if this is something that can be added to the C4IBM Web Component package, however, if not, we can look at implementing these options on C4AEM. However, the caveat here is that we'll need to extend the `cta` component and have AEM reference that file in addition to the other files that they're already consuming from C4AEM. Might have a slight performance impact, but nothing too pressing. Let me know what you both think.

### Application/website

AEM

### Business priority

Medium Priority = upcoming release but is not pressing

### What time frame would this ideally be needed by (if applicable)

- Would be great to include this change in the next release as AEM is in the midst of changing the usage of Font Family as a means to render the icon and adjust to fully adopt SVG code coming from C4IBM.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

**Comment by proeung** [comment]
@jwitkowski79 Let me know if there are other icons that you and others are using as well.

**Comment by oliviaflory** [comment]
We might want to look at opening an issue to update our CTA documentation on the website and add any additional icons required. 

For reference, these are the icons in IDL:

**Blog**
![Screen Shot 2022-05-06 at 12 07 15 PM](https://user-images.githubusercontent.com/45692622/167170806-150dba13-5f66-4ab1-8d85-8f37fa7ff125.png)




**PDF** we should discuss which is most appropriate for the use case
![Screen Shot 2022-05-06 at 12 07 32 PM](https://user-images.githubusercontent.com/45692622/167170793-3b8420db-33a0-4f46-ab22-7326b91b6b8b.png)

**Comment by oliviaflory** [comment]
Questions for Jen and Putra:

1. Are there other CTA types you know of that you’ll need? Totally ok if we need to add more later too.
2. For PDF, I was curious if a PDF would open in a new tab or if it would automatically download? 
   a. We have a download icon already, so I was thinking if it auto downloads maybe we’d be ok with that?
   b. Or we choose one of the PDF icons depending on the UX use case.

**Comment by jwitkowski79** [comment]
@oliviaflory @proeung Hello! We actually have a ticket open for this on the AEM side to update our icons [https://jsw.ibm.com/browse/AEMSITES-51](https://jsw.ibm.com/browse/AEMSITES-51). 

- We're looking to also include just a general download icon from IDL. The current PDF icon we're using in AEM (the one labeled PDF) looks odd because it's not based aligned to the link text..and it looks like it should since PDF is text. We can also use the general download icon for other file types besides pdfs. If we do have to be specific with PDF files and use one of the icons above, the "Document PDF" is probably more preferable since it's more icon like and won't look mispositioned. 
- We're also looking to include a jump link option and open in new tab option. Right now on the AEM side, we have the down arrow displaying for new tab which is incorrect.
- So to summarize, on the AEM side we'd love to include the general download icon, update the PDF icon to "Document PDF" if we have to have a specific PDF type icon, jump link, new tab, external link, video and blog.

**Comment by oliviaflory** [comment]
Adding some examples for context. I believe we have everything you listed other than
- PDF
- New tab
- Blog
![Screen Shot 2022-05-09 at 3 52 07 PM](https://user-images.githubusercontent.com/45692622/167487138-a60db19e-1bf9-47c0-83ee-bc02c8ea10ee.png)




Showing what we might see in context:

![Screen Shot 2022-05-09 at 3 52 12 PM](https://user-images.githubusercontent.com/45692622/167487148-be752ed0-10fa-47b4-aa12-66437e8eeba3.png)

![Screen Shot 2022-05-09 at 3 53 34 PM](https://user-images.githubusercontent.com/45692622/167487158-bda14a48-2ea9-4757-8a22-a08c5380c72a.png)

- I see what you mean about the `PDF` looking off, I feel alright with the document PDF option.

@jwitkowski79 New tab is one I'd love to hear more about, we've had guidelines for a while to say not to open anything in a new tab other than external links, but happy to hear what use case AEM has before saying no :)

**Comment by jwitkowski79** [comment]
@oliviaflory You are correct on the new tab icon! However, let me check with Joseph on this as there is a "new tab" option in AEM for links, which we just may need to strip out on the AEM side.

**Comment by EAVilleda** [comment]
I just want to add that we have always had provided authors the option to open any link in a new tab by providing a "Target" field in the AEM component dialogs. The "New Tab" option never affects the icons, the type of link is what affects the icon. In the case Jen described above, the link had a down arrow because having # as the link acts as an anchor.

**Comment by oliviaflory** [comment]
Thanks @jwitkowski79 and @EAVilleda, The down arrow is associated with the current anchor link visual styling 👍 

Our teams should discuss opening things in new tabs so we are aligned on the implementation and guidelines. In general, all links should open within the same tab, unless the user is being directed outside of the IBM domain, in which case it should use the Launch icon paired with the Leaving IBM modal.
 
These guidelines are documented on the [web standards](https://w3.ibm.com/w3publisher/ibm-web-standards-external/standards-mandatory/page-layout-design) site and follow [w3](https://www.w3.org/TR/WCAG20-TECHS/G200.html) and [Nielsen Normal](https://www.nngroup.com/articles/new-browser-windows-and-tabs/) best practices. We want to avoid having different linking experiences across various pages.

**Comment by jwitkowski79** [comment]
@oliviaflory I completely agree. We also have documentation on the AEM side about this. I'm checking to see if we need to include the "new tab" icon in AEM for those outlier occasions where something does open in a new tab. If not, we can remove the new tab icon since the External" icon would cover it

**Comment by jwitkowski79** [comment]
@oliviaflory Hello! So our team met on the icons and here's what we had discussed.  We are proposing to have the following CTA icons in AEM:
- Local
- Jump
- External (leaving IBM)
- New Tab
- Download
- Video
- Document PDF

We are proposing to remove the following icons:
- Blog
- PDF

![Screen Shot 2022-05-17 at 2 17 11 PM](https://user-images.githubusercontent.com/2040791/168884352-5883c5d0-4aa0-4410-9a6e-e415c4d1233b.png)

For the New Tab icon, we completely agree that links should not open in a new tab, unless the user is leaving IBM. However, we want to keep it as a backup. It would rarely get used but a couple of good points came up about what if a link takes a long time to load...say a link to a demo. It probably wouldn't even get used in that case, however we need some time to explore if it's truly wise to eliminate this icon all together.

For the blog icon, we're also going to remove this as our team thinks it's confusing and using the local link icon would work just fine. 

For the PDF icon, we'd like to only include the Document PDF because the other text based PDF icon has some vertical alignment issues with the text. It's being vertically centered but because it's a text based icon, it looks like it should be based aligned to the text.

Let me know what you think with the above!

**Comment by kjwhitla** [comment]
@jwitkowski79 @oliviaflory  
A couple of thoughts - 
- As far as opening in a new window/same tab, in AEM we will be keeping the ability as necessary to support the functionality of the different windows. Content guidance for authors should be the point of control for the experience.
- Although design-wise we may want to immediately/truly remove the Blog/PDF CTA Icons from AEM, Is this another case where instead want to control this with content guidance?

**Comment by jwitkowski79** [comment]
@proeung @oliviaflory @kjwhitla Here's a new screenshot of the proposed icons in AEM. I think it makes sense to keep "blog" and just control with content guidance. However, we think having 2 pdf icons is confusing. We'd just like to remove the text version of the pdf icon and replace it with the document version of the icon. See attached. 

@oliviaflory The content guidance of when to use a new tab will follow what's published on the web standards page. We don't see this being used, but we want to keep it in the system for those rare instances. 
![Frame 196](https://user-images.githubusercontent.com/2040791/170522084-9c6f0a34-9c1c-4bd1-8068-f7049c3edf75.jpg)

**Comment by proeung** [comment]
@jwitkowski79 Thanks!

@oliviaflory @kennylam Feels like we have enough info to bring this issue to the development pipeline. Let me know if this is something that our team can take a look at this week. Thanks!

**Comment by IgnacioBecerra** [comment]
@Gopi916 Staging storybook was slightly outdated -- it should now be testable on staging.
